api/v0/pb/provider.proto

```diff
@@ -0,0 +1,23 @@
+// In order to re-generate the golang packages for `Message` you will need...
+// 1. Protobuf binary (tested with protoc 3.0.0). - https://github.com/gogo/protobuf/releases
+// 2. Gogo Protobuf (tested with gogo 0.3). - https://github.com/gogo/protobuf
+// Now from `libp2p/<path>/pb` you can run...
+// `protoc --gogo_out=. --proto_path=../../<path>/pb/ --proto_path=./ --proto_path=/usr/include pb.proto`
+
+syntax = "proto3";
+package reqresp.pb;
+
+message ProviderMessage {
+    enum MessageType {
```

Comment:
Flexibility? Ease of implementation? I just keep seeing this approach for libp2p protocols and I have internalized it. See for example https://github.com/libp2p/go-libp2p-kad-dht/blob/master/pb/dht.proto.

Reply:
OK thanks. this seemed slightly odd to me, since we pay the cost of depending on protobuf but don't take full advantage of what it can offer: backward compatibility, contract-first design, message schema conformity etc.